src/TimerUnit.cpp

```diff
@@ -301,14 +311,14 @@ bool TimerUnit::disableTimer(const QString& name)
     return found;
 }
 
-TTimer* TimerUnit::findTimer(const QString& name)
+// This is currently only used during the lua scripted creation of a new
+// permTime to find a parent with the given name:
+QList<TTimer*> TimerUnit::findTimer(const QString& name)
 {
-    QMap<QString, TTimer*>::const_iterator it = mLookupTable.constFind(name);
-    while (it != mLookupTable.cend() && it.key() == name) {
-        TTimer* pT = it.value();
-        return pT;
-    }
-    return nullptr;
+    // Previously a constFind(name) STL was being used which returns the most
+    // recently added element with a given key first - we now return all the
+    // matches but the first is still the most recently added:
+    return mLookupTable.values(name);
```

Comment:
Kind of inefficient to create a whole new list every time you want to find a timer just to use the lists first entry only, no?

Reply:
Depends on what the caller wanted - the list may be empty, contain just one entry (we hope) or multiples - this does allow the upstream (Lua code) what to do with those different cases - as it is the code that looks for a parent is flawed in that it **always** uses the first one found - which may not be what the end user wants. TBH We ought to allow them to set a parent by ID given that that is what we often return to them.  Besides, if it is a list of pointers to Qt Objects to be returned the construction costs are not going to be *too* bad are they?

Reply:
But this is based on theory though, not an actual problem we've had?

I'm just keen on having performance at the front, if we don't need to do inefficient things we shouldn't.

Reply:
Our lua API is already *inefficient* in that users probably do not yet realise yet what duplicate names might mean (that only the first one is accessible to be a parent). I've already found this unawareness might a problem when I found out that key-bindings were potentially being ignored in #1656!

We might be able to improve it - at present we do report if there is no parent timer with the given name - which both this and the prior code handles but we could expand the corresponding functions to accept an index to select the *wanted* parent (if `èxists(name, type)` returns a result greater than `1`). In which case the lua function that calls this one will want a list, not just a pointer / `nullptr`. That can, though, be left for another PR.

Remember this is only used during the *creation* of the timer - not when it is *firing*, so it does not seem to me to be something that is critical in the executing process.

Reply:
Nobody's complaining, so it must be working well... 

1 timer creation = 1 timer fire from the Lua API, so it needs to be fast in both! There's no reason to change it from returning a single timer to creating a list, putting one timer into it, returning the list, and then only actually using the first item in the list :thinking:

Reply:
Let's have both methods perhaps?

Reply:
Okay - I'll redo the code in the `TTimerUnit` class; and leave an expansion of the lua API to handle multiple potential parents when we have something that can be applied uniformly across all the `perm(Alias|Key|Timer|`\<several trigger types>`)(...)` types that may have this issue.

Reply:
Done.